benchmark compile error

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
benchmark compile error 
```
[100%] Built target orc_column_reader_bench
In file included from /root/starrocks/be/src/common/cow.h:23,
                 from /root/starrocks/be/src/column/column.h:26,
                 from /root/starrocks/be/src/column/const_column.h:17,
                 from /root/starrocks/be/src/column/column_helper.h:22,
                 from /root/starrocks/be/src/bench/get_dict_codes_bench.cpp:20:
/root/starrocks/be/src/gutil/casts.h: In instantiation of ‘To down_cast(From*) [with To = starrocks::BinaryColumnBase<unsigned int>*; From = const starrocks::Column]’:
/root/starrocks/be/src/bench/get_dict_codes_bench.cpp:89:66:   required from here
   89 |         auto* dict_value_binary_column = down_cast<BinaryColumn*>(dict_nullable_column->data_column().get());
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/starrocks/be/src/gutil/casts.h:80:12: error: ‘static_cast’ from type ‘const starrocks::Column*’ to type ‘starrocks::BinaryColumnBase<unsigned int>*’ casts away qualifiers
   80 |     return static_cast<To>(f);
      |            ^~~~~~~~~~~~~~~~~~
make[2]: *** [src/bench/CMakeFiles/get_dict_codes_bench.dir/build.make:79: src/bench/CMakeFiles/get_dict_codes_bench.dir/get_dict_codes_bench.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2383: src/bench/CMakeFiles/get_dict_codes_bench.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

/root/starrocks/be/src/bench/hash_functions_bench.cpp: In member function ‘void starrocks::HourFromUnixtimeBench::bench_from_unixtime_extract_hour(benchmark::State&)’:
/root/starrocks/be/src/bench/hash_functions_bench.cpp:247:47: error: ‘class starrocks::Column’ has no member named ‘get_data’
  247 |                 dt_col->as_mutable_raw_ptr()->get_data()[i].to_timestamp(&year, &month, &day, &hour, &minute, &second,
      |                                               ^~~~~~~~
make[2]: *** [src/bench/CMakeFiles/hash_functions_bench.dir/build.make:79: src/bench/CMakeFiles/hash_functions_bench.dir/hash_functions_bench.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2563: src/bench/CMakeFiles/hash_functions_bench.dir/all] Error 2
/root/starrocks/be/src/bench/object_cache_bench.cpp: In member function ‘void starrocks::ObjectCacheBench::init_cache(starrocks::CacheType)’:
/root/starrocks/be/src/bench/object_cache_bench.cpp:123:38: error: cannot convert ‘starrocks::DiskCacheOptions’ to ‘const starrocks::MemCacheOptions&’
  123 |         Status st = _lru_cache->init(opt);
      |                                      ^~~
      |                                      |
      |                                      starrocks::DiskCacheOptions
In file included from /root/starrocks/be/src/bench/object_cache_bench.cpp:22:
/root/starrocks/be/src/cache/mem_cache/lrucache_engine.h:28:40: note:   initializing argument 1 of ‘starrocks::Status starrocks::LRUCacheEngine::init(const starrocks::MemCacheOptions&)’
   28 |     Status init(const MemCacheOptions& options);
      |                 ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/root/starrocks/be/src/bench/object_cache_bench.cpp:137:42: error: cannot convert ‘std::__shared_ptr<starrocks::StarCacheEngine, __gnu_cxx::_S_atomic>::element_type*’ {aka ‘starrocks::StarCacheEngine*’} to ‘starrocks::LocalMemCacheEngine*’
  137 |         _page_cache->init(_star_cache.get());
      |                           ~~~~~~~~~~~~~~~^~
      |                                          |
      |                                          std::__shared_ptr<starrocks::StarCacheEngine, __gnu_cxx::_S_atomic>::element_type* {aka starrocks::StarCacheEngine*}
In file included from /root/starrocks/be/src/bench/object_cache_bench.cpp:23:
/root/starrocks/be/src/cache/mem_cache/page_cache.h:71:36: note:   initializing argument 1 of ‘void starrocks::StoragePageCache::init(starrocks::LocalMemCacheEngine*)’
   71 |     void init(LocalMemCacheEngine* cache_engine) {
      |               ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
make[2]: *** [src/bench/CMakeFiles/object_cache_bench.dir/build.make:79: src/bench/CMakeFiles/object_cache_bench.dir/object_cache_bench.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2983: src/bench/CMakeFiles/object_cache_bench.dir/all] Error 2
In file included from /opt/gcc-toolset-14/include/c++/14.3.0/bits/stl_iterator.h:78,
                 from /opt/gcc-toolset-14/include/c++/14.3.0/bits/stl_algobase.h:67,
                 from /opt/gcc-toolset-14/include/c++/14.3.0/algorithm:60,
                 from /var/local/thirdparty/installed/include/benchmark/benchmark.h:179,
                 from /root/starrocks/be/src/bench/chunks_sorter_bench.cpp:15:
/opt/gcc-toolset-14/include/c++/14.3.0/bits/stl_construct.h: In instantiation of ‘constexpr void std::_Construct(_Tp*, _Args&& ...) [with _Tp = starrocks::Chunk; _Args = {vector<starrocks::Cow<starrocks::Column>::ImmutPtr<starrocks::Column>, allocator<starrocks::Cow<starrocks::Column>::ImmutPtr<starrocks::Column> > >&, phmap::flat_hash_map<int, long unsigned int, starrocks::StdHash<int>, phmap::EqualTo<int>, allocator<pair<const int, long unsigned int> > >&}]’:
/opt/gcc-toolset-14/include/c++/14.3.0/bits/alloc_traits.h:706:19:   required from ‘static constexpr void std::allocator_traits<std::allocator<void> >::construct(allocator_type&, _Up*, _Args&& ...) [with _Up = starrocks::Chunk; _Args = {std::vector<starrocks::Cow<starrocks::Column>::ImmutPtr<starrocks::Column>, std::allocator<starrocks::Cow<starrocks::Column>::ImmutPtr<starrocks::Column> > >&, phmap::flat_hash_map<int, long unsigned int, starrocks::StdHash<int>, phmap::EqualTo<int>, std::allocator<std::pair<const int, long unsigned int> > >&}; allocator_type = std::allocator<void>]’
  706 |         { std::_Construct(__p, std::forward<_Args>(__args)...); }
      |           ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/gcc-toolset-14/include/c++/14.3.0/bits/shared_ptr_base.h:607:39:   required from ‘std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_Sp_counted_ptr_inplace(_Alloc, _Args&& ...) [with _Args = {std::vector<starrocks::Cow<starrocks::Column>::ImmutPtr<starrocks::Column>, std::allocator<starrocks::Cow<starrocks::Column>::ImmutPtr<starrocks::Column> > >&, phmap::flat_hash_map<int, long unsigned int, starrocks::StdHash<int>, phmap::EqualTo<int>, std::allocator<std::pair<const int, long unsigned int> > >&}; _Tp = starrocks::Chunk; _Alloc = std::allocator<void>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
  607 |           allocator_traits<_Alloc>::construct(__a, _M_ptr(),
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
  608 |               std::forward<_Args>(__args)...); // might throw
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
/opt/gcc-toolset-14/include/c++/14.3.0/bits/shared_ptr_base.h:969:16:   required from ‘std::__shared_count<_Lp>::__shared_count(_Tp*&, std::_Sp_alloc_shared_tag<_Alloc>, _Args&& ...) [with _Tp = starrocks::Chunk; _Alloc = std::allocator<void>; _Args = {std::vector<starrocks::Cow<starrocks::Column>::ImmutPtr<starrocks::Column>, std::allocator<starrocks::Cow<starrocks::Column>::ImmutPtr<starrocks::Column> > >&, phmap::flat_hash_map<int, long unsigned int, starrocks::StdHash<int>, phmap::EqualTo<int>, std::allocator<std::pair<const int, long unsigned int> > >&}; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
  969 |           auto __pi = ::new (__mem)
      |                       ^~~~~~~~~~~~~
  970 |             _Sp_cp_type(__a._M_a, std::forward<_Args>(__args)...);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Expected behavior (Required)

### Real behavior (Required)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

--- a/be/src/bench/chunks_sorter_bench.cpp
+++ b/be/src/bench/chunks_sorter_bench.cpp
@@ -219,9 +219,8 @@ static void do_bench(benchmark::State& state, SortAlgorithm sorter_algo, Logical
         sort_exprs.push_back(sort_expr);
         asc_arr.push_back(true);
         null_first.push_back(true);
-        map[i] = i;
     }
-    auto chunk = std::make_shared<Chunk>(columns, map);
+    auto chunk = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
 
     RuntimeState* runtime_state = suite._runtime_state.get();
     int64_t item_processed = 0;
@@ -334,9 +333,8 @@ static void do_heap_merge(benchmark::State& state, int num_runs, bool use_merger
 
         asc_arr.push_back(true);
         null_first.push_back(true);
-        map[i] = i;
     }
-    ChunkPtr base_chunk = std::make_shared<Chunk>(columns, map);
+    ChunkPtr base_chunk = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
 
     int64_t num_rows = 0;
     for (auto _ : state) {
@@ -418,10 +416,9 @@ static void do_merge_columnwise(benchmark::State& state, int num_runs, bool null
         sort_exprs.push_back(sort_expr);
         asc_arr.push_back(true);
         null_first.push_back(true);
-        map[i] = i;
     }
-    ChunkPtr chunk1 = std::make_shared<Chunk>(columns, map);
-    ChunkPtr chunk2 = std::make_shared<Chunk>(columns, map);
+    ChunkPtr chunk1 = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
+    ChunkPtr chunk2 = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
 
     int64_t num_rows = 0;
     SortDescs sort_desc(std::vector<int>{1, 1, 1}, std::vector<int>{-1, -1, -1});
@@ -468,9 +465,8 @@ static void do_bench_materialize(benchmark::State& state, LogicalType data_type,
     for (int i = 0; i < num_columns; i++) {
         auto [column, expr] = suite.build_column(type_desc, i, false, nullable);
         columns.push_back(column);
-        map[i] = i;
     }
-    auto template_chunk = std::make_shared<Chunk>(columns, map);
+    auto template_chunk = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
 
     std::vector<ChunkPtr> chunks;
     std::vector<PermutationItem> perm;

--- a/be/src/bench/chunks_sorter_bench.cpp
+++ b/be/src/bench/chunks_sorter_bench.cpp
@@ -219,8 +219,9 @@ static void do_bench(benchmark::State& state, SortAlgorithm sorter_algo, Logical
         sort_exprs.push_back(sort_expr);
         asc_arr.push_back(true);
         null_first.push_back(true);
+        map[i] = i;
     }
-    auto chunk = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
+    auto chunk = std::make_shared<Chunk>(std::move(columns), map);
 
     RuntimeState* runtime_state = suite._runtime_state.get();
     int64_t item_processed = 0;
@@ -333,8 +334,9 @@ static void do_heap_merge(benchmark::State& state, int num_runs, bool use_merger
 
         asc_arr.push_back(true);
         null_first.push_back(true);
+        map[i] = i;
     }
-    ChunkPtr base_chunk = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
+    ChunkPtr base_chunk = std::make_shared<Chunk>(std::move(columns), map);
 
     int64_t num_rows = 0;
     for (auto _ : state) {
@@ -416,9 +418,10 @@ static void do_merge_columnwise(benchmark::State& state, int num_runs, bool null
         sort_exprs.push_back(sort_expr);
         asc_arr.push_back(true);
         null_first.push_back(true);
+        map[i] = i;
     }
-    ChunkPtr chunk1 = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
-    ChunkPtr chunk2 = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
+    ChunkPtr chunk1 = std::make_shared<Chunk>(columns, map);
+    ChunkPtr chunk2 = std::make_shared<Chunk>(std::move(columns), map);
 
     int64_t num_rows = 0;
     SortDescs sort_desc(std::vector<int>{1, 1, 1}, std::vector<int>{-1, -1, -1});
@@ -465,8 +468,9 @@ static void do_bench_materialize(benchmark::State& state, LogicalType data_type,
     for (int i = 0; i < num_columns; i++) {
         auto [column, expr] = suite.build_column(type_desc, i, false, nullable);
         columns.push_back(column);
+        map[i] = i;
     }
-    auto template_chunk = std::make_shared<Chunk>(columns, Chunk::SlotHashMap{});
+    auto template_chunk = std::make_shared<Chunk>(std::move(columns), map);
 
     std::vector<ChunkPtr> chunks;
     std::vector<PermutationItem> perm;

--- a/be/src/bench/get_dict_codes_bench.cpp
+++ b/be/src/bench/get_dict_codes_bench.cpp
@@ -86,7 +86,7 @@ static void BM_GetDictCodesWithMap(benchmark::State& state) {
         }
 
         auto* dict_nullable_column = down_cast<NullableColumn*>(column.get());
-        auto* dict_value_binary_column = down_cast<BinaryColumn*>(dict_nullable_column->data_column().get());
+        auto* dict_value_binary_column = down_cast<const BinaryColumn*>(dict_nullable_column->data_column().get());
         auto dict_values_filtered = dict_value_binary_column->get_data();
         if (!has_null) {
             dict_codes.reserve(dict_values_filtered.size());

--- a/be/src/bench/get_dict_codes_bench.cpp
+++ b/be/src/bench/get_dict_codes_bench.cpp
@@ -86,7 +86,8 @@ static void BM_GetDictCodesWithMap(benchmark::State& state) {
         }
 
         auto* dict_nullable_column = down_cast<NullableColumn*>(column.get());
-        auto* dict_value_binary_column = down_cast<const BinaryColumn*>(dict_nullable_column->data_column().get());
+        const auto* dict_value_binary_column =
+                down_cast<const BinaryColumn*>(dict_nullable_column->data_column().get());
         auto dict_values_filtered = dict_value_binary_column->get_data();
         if (!has_null) {
             dict_codes.reserve(dict_values_filtered.size());

--- a/be/src/bench/hash_functions_bench.cpp
+++ b/be/src/bench/hash_functions_bench.cpp
@@ -241,11 +241,13 @@ public:
             auto dt_col_ptr = starrocks::TimeFunctions::from_unix_to_datetime_64(utils->get_fn_ctx(), columns).value();
             auto dt_col = starrocks::ColumnHelper::cast_to<starrocks::TYPE_DATETIME>(dt_col_ptr);
             int64_t sum = 0;
-            // Use const_cast for read-only benchmark access
+
+            auto* timestamp_col = down_cast<TimestampColumn*>(dt_col.get());
+
+            // Read-only benchmark access over TimestampColumn data
             for (int i = 0; i < N; ++i) {
                 int year, month, day, hour, minute, second, usec;
-                dt_col->as_mutable_raw_ptr()->get_data()[i].to_timestamp(&year, &month, &day, &hour, &minute, &second,
-                                                                         &usec);
+                timestamp_col->get_data()[i].to_timestamp(&year, &month, &day, &hour, &minute, &second, &usec);
                 sum += hour;
             }
             benchmark::DoNotOptimize(sum);

--- a/be/src/bench/hash_functions_bench.cpp
+++ b/be/src/bench/hash_functions_bench.cpp
@@ -242,7 +242,7 @@ public:
             auto dt_col = starrocks::ColumnHelper::cast_to<starrocks::TYPE_DATETIME>(dt_col_ptr);
             int64_t sum = 0;
 
-            auto* timestamp_col = down_cast<TimestampColumn*>(dt_col.get());
+            const auto* timestamp_col = down_cast<const TimestampColumn*>(dt_col.get());
 
             // Read-only benchmark access over TimestampColumn data
             for (int i = 0; i < N; ++i) {

--- a/be/src/bench/object_cache_bench.cpp
+++ b/be/src/bench/object_cache_bench.cpp
@@ -17,6 +17,7 @@
 #include <cstdlib>
 #include <random>
 
+#include "cache/block_cache/block_cache.h"
 #include "cache/cache_options.h"
 #include "cache/disk_cache/starcache_engine.h"
 #include "cache/mem_cache/lrucache_engine.h"
@@ -63,6 +64,14 @@ public:
     static void random_insert_multi_threads(benchmark::State* state, StoragePageCache* cache, size_t count,
                                             size_t page_size);
 
+    void prepare_data(BlockCache* cache, int64_t count);
+    void prepare_sequence_data(BlockCache* cache, int64_t count);
+    void random_query(benchmark::State& state, BlockCache* cache, size_t ratio, int64_t iter_count, int64_t count);
+    static void random_query_multi_threads_block(benchmark::State* state, BlockCache* cache, size_t ratio, size_t count,
+                                                 size_t page_size);
+    static void random_insert_multi_threads_block(benchmark::State* state, BlockCache* cache, size_t count,
+                                                  size_t page_size);
+
     void insert_to_cache(benchmark::State& state, CacheType cache_type, int64_t count);
     void random_query(benchmark::State& state, CacheType cache_type, size_t ratio, int64_t iter_count, int64_t count);
     void random_query_multi_threads_test(benchmark::State& state, CacheType cache_type, size_t ratio, int64_t count);
@@ -73,6 +82,7 @@ private:
     size_t _capacity = 100L * 1024 * 1024 * 1024;
     size_t _page_size = 1024;
 
+    std::shared_ptr<BlockCache> _block_cache;
     std::shared_ptr<LRUCacheEngine> _lru_cache;
     std::shared_ptr<StarCacheEngine> _star_cache;
     std::shared_ptr<StoragePageCache> _page_cache;
@@ -135,9 +145,11 @@ void ObjectCacheBench::init_cache(CacheType cache_type) {
         if (!st.ok()) {
             LOG(FATAL) << "init star cache failed: " << st;
         }
-        _page_cache = std::make_shared<StoragePageCache>();
-        // _page_cache->init(_star_cache.get());
-        LOG(WARNING) << "StarCacheEngine integration disabled in benchmark due to API mismatch";
+        BlockCacheOptions block_opt;
+        block_opt.block_size = _page_size;
+        _block_cache = std::make_shared<BlockCache>();
+        _block_cache->init(block_opt, _star_cache, nullptr);
+        LOG(INFO) << "init star cache (block cache) success";
     }
 }
 
@@ -220,7 +232,79 @@ void ObjectCacheBench::random_insert_multi_threads(benchmark::State* state, Stor
             }
         }
     }
+    state->PauseTiming();
+}
 
+void ObjectCacheBench::prepare_sequence_data(BlockCache* cache, int64_t count) {
+    std::vector<uint8_t> buf(_page_size, 1);
+    for (size_t i = 0; i < count; i++) {
+        std::string key = "str:" + std::to_string(rand() % count);
+        Status st = cache->write_buffer(key, 0, _page_size, buf.data());
+        if (!st.ok()) {
+            if (!st.is_already_exist()) {
+                LOG(FATAL) << "write_buffer failed: " << st;
+            }
+        }
+    }
+}
+
+void ObjectCacheBench::prepare_data(BlockCache* cache, int64_t count) {
+    std::vector<uint8_t> buf(_page_size, 1);
+    for (size_t i = 0; i < count; i++) {
+        std::string key = "str:" + std::to_string(rand());
+        Status st = cache->write_buffer(key, 0, _page_size, buf.data());
+        if (!st.ok()) {
+            if (!st.is_already_exist()) {
+                LOG(FATAL) << "write_buffer failed: " << st;
+            }
+        }
+    }
+}
+
+void ObjectCacheBench::random_query(benchmark::State& state, BlockCache* cache, size_t ratio, int64_t iter_count,
+                                    int64_t count) {
+    thread_local std::mt19937 gen(std::random_device{}());
+    thread_local std::uniform_int_distribution<> dis(1, 1073741824);
+    std::vector<char> buf(_page_size);
+
+    LOG(ERROR) << "random query start";
+    for (size_t i = 0; i < iter_count; i++) {
+        std::string key = "str:" + std::to_string(dis(gen) % (count * ratio));
+        (void)cache->read_buffer(key, 0, _page_size, buf.data());
+    }
+    LOG(ERROR) << "random query end";
+}
+
+void ObjectCacheBench::random_query_multi_threads_block(benchmark::State* state, BlockCache* cache, size_t ratio,
+                                                        size_t count, size_t page_size) {
+    state->ResumeTiming();
+    thread_local std::mt19937 gen(std::random_device{}());
+    thread_local std::uniform_int_distribution<> dis(1, 1073741824);
+    std::vector<char> buf(page_size);
+
+    for (size_t i = 0; i < count; i++) {
+        std::string key = "str:" + std::to_string(dis(gen) % (count * ratio));
+        (void)cache->read_buffer(key, 0, page_size, buf.data());
+    }
+    state->PauseTiming();
+}
+
+void ObjectCacheBench::random_insert_multi_threads_block(benchmark::State* state, BlockCache* cache, size_t count,
+                                                         size_t page_size) {
+    state->ResumeTiming();
+    thread_local std::mt19937 gen(std::random_device{}());
+    thread_local std::uniform_int_distribution<> dis(1, 1073741824);
+    std::vector<uint8_t> buf(page_size, 1);
+
+    for (size_t i = 0; i < count; i++) {
+        std::string key = "str:" + std::to_string(dis(gen));
+        Status st = cache->write_buffer(key, 0, page_size, buf.data());
+        if (!st.ok()) {
+            if (!st.is_already_exist()) {
+                LOG(FATAL) << "write_buffer failed: " << st;
+            }
+        }
+    }
     state->PauseTiming();
 }
 
@@ -230,11 +314,20 @@ void ObjectCacheBench::insert_to_cache(benchmark::State& state, CacheType cache_
     init_cache(cache_type);
 
     state.ResumeTiming();
-    prepare_data(_page_cache.get(), count);
+    if (cache_type == CacheType::LRU) {
+        prepare_data(_page_cache.get(), count);
+    } else {
+        prepare_data(_block_cache.get(), count);
+    }
     state.PauseTiming();
 
     int64_t new_mem_usage = CurrentThread::mem_tracker()->consumption();
-    int64_t calc_usage = _page_cache->memory_usage() / 1024 / 1024;
+    int64_t calc_usage = 0;
+    if (cache_type == CacheType::LRU) {
+        calc_usage = _page_cache->memory_usage() / 1024 / 1024;
+    } else {
+        calc_usage = _star_cache->mem_usage() / 1024 / 1024;
+    }
     int64_t real_usage = (new_mem_usage - old_mem_usage) / 1024 / 1024;
 
     LOG(INFO) << "insert: type=" << type_str << ", metric=" << calc_usage << "M, lru=" << real_usage << "M";
@@ -243,10 +336,15 @@ void ObjectCacheBench::insert_to_cache(benchmark::State& state, CacheType cache_
 void ObjectCacheBench::random_query(benchmark::State& state, CacheType cache_type, size_t ratio, int64_t iter_count,
                                     int64_t count) {
     init_cache(cache_type);
-    prepare_sequence_data(_page_cache.get(), count);
-
-    state.ResumeTiming();
-    random_query(state, _page_cache.get(), ratio, iter_count, count);
+    if (cache_type == CacheType::LRU) {
+        prepare_sequence_data(_page_cache.get(), count);
+        state.ResumeTiming();
+        random_query(state, _page_cache.get(), ratio, iter_count, count);
+    } else {
+        prepare_sequence_data(_block_cache.get(), count);
+        state.ResumeTiming();
+        random_query(state, _block_cache.get(), ratio, iter_count, count);
+    }
     state.PauseTiming();
 }
 
@@ -254,18 +352,32 @@ void ObjectCacheBench::random_query_multi_threads_test(benchmark::State& state, 
                                                        int64_t count) {
     state.PauseTiming();
     init_cache(cache_type);
-    prepare_sequence_data(_page_cache.get(), count);
 
     LOG(INFO) << "start random query test";
     std::vector<std::thread> threads;
-    for (size_t i = 0; i < 10; i++) {
-        threads.emplace_back(random_query_multi_threads, &state, _page_cache.get(), ratio, count);
+    if (cache_type == CacheType::LRU) {
+        prepare_sequence_data(_page_cache.get(), count);
+        for (size_t i = 0; i < 10; i++) {
+            threads.emplace_back(random_query_multi_threads, &state, _page_cache.get(), ratio, count);
+        }
+    } else {
+        prepare_sequence_data(_block_cache.get(), count);
+        for (size_t i = 0; i < 10; i++) {
+            threads.emplace_back(random_query_multi_threads_block, &state, _block_cache.get(), ratio, count,
+                                 _page_size);
+        }
     }
+
     for (auto& t : threads) {
         t.join();
     }
-    LOG(INFO) << "end random query test: lookup=" << _page_cache->get_lookup_count()
-              << ", hit=" << _page_cache->get_hit_count();
+
+    if (cache_type == CacheType::LRU) {
+        LOG(INFO) << "end random query test: lookup=" << _page_cache->get_lookup_count()
+                  << ", hit=" << _page_cache->get_hit_count();
+    } else {
+        LOG(INFO) << "end random query test (BlockCache stats not directly available in this bench)";
+    }
 
     threads.clear();
     state.ResumeTiming();
@@ -278,7 +390,11 @@ void ObjectCacheBench::insert_cache_multi_threads_test(benchmark::State& state, 
 
     std::vector<std::thread> threads;
     for (size_t i = 0; i < 5; i++) {
-        threads.emplace_back(random_insert_multi_threads, &state, _page_cache.get(), count, _page_size);
+        if (cache_type == CacheType::LRU) {
+            threads.emplace_back(random_insert_multi_threads, &state, _page_cache.get(), count, _page_size);
+        } else {
+            threads.emplace_back(random_insert_multi_threads_block, &state, _block_cache.get(), count, _page_size);
+        }
     }
     for (auto& t : threads) {
         t.join();
@@ -324,25 +440,23 @@ static void bench_func(benchmark::State& state) {
 static void process_args(benchmark::internal::Benchmark* b) {
     // one thread insert
     b->Args({0, 0, 10000000, 10000000})->Iterations(1);
-    // StarCache (Type 1) tests disabled due to broken integration with StoragePageCache
-    // TODO: Re-enable these tests once StarCacheEngine integration is fixed.
-    // b->Args({1, 0, 10000000, 10000000})->Iterations(1);
+    b->Args({1, 0, 10000000, 10000000})->Iterations(1);
 
     // one thread query, all hit
     b->Args({0, 1, 10000000, 10000000})->Iterations(1);
-    // b->Args({1, 1, 10000000, 10000000})->Iterations(1);
+    b->Args({1, 1, 10000000, 10000000})->Iterations(1);
 
     // one thread query, 50% hit
     b->Args({0, 2, 10000000, 10000000})->Iterations(1);
-    // b->Args({1, 2, 10000000, 10000000})->Iterations(1);
+    b->Args({1, 2, 10000000, 10000000})->Iterations(1);
 
     // multi thread query
     b->Args({0, 3, 10000000, 10000000})->Iterations(1);
-    // b->Args({1, 3, 10000000, 10000000})->Iterations(1);
+    b->Args({1, 3, 10000000, 10000000})->Iterations(1);
 
     // multi thread insert
     b->Args({0, 4, 1000000, 5000000})->Iterations(1);
-    // b->Args({1, 4, 1000000, 5000000})->Iterations(1);
+    b->Args({1, 4, 1000000, 5000000})->Iterations(1);
 }
 
 BENCHMARK(bench_func)->Apply(process_args);

--- a/be/src/bench/object_cache_bench.cpp
+++ b/be/src/bench/object_cache_bench.cpp
@@ -120,9 +120,11 @@ void ObjectCacheBench::init_cache(CacheType cache_type) {
 
     if (cache_type == CacheType::LRU) {
         _lru_cache = std::make_shared<LRUCacheEngine>();
-        Status st = _lru_cache->init(opt);
+        MemCacheOptions mem_opt;
+        mem_opt.capacity = _capacity;
+        Status st = _lru_cache->init(mem_opt);
         if (!st.ok()) {
-            LOG(FATAL) << "init star cache failed: " << st;
+            LOG(FATAL) << "init lru cache failed: " << st;
         }
         LOG(INFO) << "init lru cache success";
         _page_cache = std::make_shared<StoragePageCache>();
@@ -134,8 +136,8 @@ void ObjectCacheBench::init_cache(CacheType cache_type) {
             LOG(FATAL) << "init star cache failed: " << st;
         }
         _page_cache = std::make_shared<StoragePageCache>();
-        _page_cache->init(_star_cache.get());
-        LOG(INFO) << "init star cache succ";
+        // _page_cache->init(_star_cache.get());
+        LOG(WARNING) << "StarCacheEngine integration disabled in benchmark due to API mismatch";
     }
 }
 
@@ -322,23 +324,25 @@ static void bench_func(benchmark::State& state) {
 static void process_args(benchmark::internal::Benchmark* b) {
     // one thread insert
     b->Args({0, 0, 10000000, 10000000})->Iterations(1);
-    b->Args({1, 0, 10000000, 10000000})->Iterations(1);
+    // StarCache (Type 1) tests disabled due to broken integration with StoragePageCache
+    // TODO: Re-enable these tests once StarCacheEngine integration is fixed.
+    // b->Args({1, 0, 10000000, 10000000})->Iterations(1);
 
     // one thread query, all hit
     b->Args({0, 1, 10000000, 10000000})->Iterations(1);
-    b->Args({1, 1, 10000000, 10000000})->Iterations(1);
+    // b->Args({1, 1, 10000000, 10000000})->Iterations(1);
 
     // one thread query, 50% hit
     b->Args({0, 2, 10000000, 10000000})->Iterations(1);
-    b->Args({1, 2, 10000000, 10000000})->Iterations(1);
+    // b->Args({1, 2, 10000000, 10000000})->Iterations(1);
 
     // multi thread query
     b->Args({0, 3, 10000000, 10000000})->Iterations(1);
-    b->Args({1, 3, 10000000, 10000000})->Iterations(1);
+    // b->Args({1, 3, 10000000, 10000000})->Iterations(1);
 
     // multi thread insert
     b->Args({0, 4, 1000000, 5000000})->Iterations(1);
-    b->Args({1, 4, 1000000, 5000000})->Iterations(1);
+    // b->Args({1, 4, 1000000, 5000000})->Iterations(1);
 }
 
 BENCHMARK(bench_func)->Apply(process_args);

--- a/be/src/bench/orc_column_reader_bench.cpp
+++ b/be/src/bench/orc_column_reader_bench.cpp
@@ -251,7 +251,7 @@ static void BM_primitive(benchmark::State& state) {
     orcChunkReader.disable_broker_load_mode();
 
     std::unique_ptr<ORCColumnReader> orcColumnReader =
-            ORCColumnReader::create(c0Type, orcType, isNullable, orcMapping, &orcChunkReader).value();
+            ORCColumnReader::create(c0Type, orcType, isNullable, orcMapping.get(), &orcChunkReader).value();
 
     for (auto _ : state) {
         ORC_UNIQUE_PTR<orc::ColumnVectorBatch> batch = rr->createRowBatch(columnSize);
@@ -261,7 +261,7 @@ static void BM_primitive(benchmark::State& state) {
         size_t totalNumRows = 0;
         while (rr->next(*batch, &pos)) {
             ColumnPtr column = ColumnHelper::create_column(c0Type, isNullable);
-            CHECK(orcColumnReader->get_next(c0, column, 0, columnSize).ok());
+            CHECK(orcColumnReader->get_next(c0, column.get(), 0, columnSize).ok());
             DCHECK_EQ(columnSize, column->size());
             totalNumRows += columnSize;
         }

